src/components/DraggableList/DraggableList.jsx

```diff
@@ -0,0 +1,226 @@
+
+const DraggableList = createClass({
+	statics: {
+		peek: {
+			description: `
+				This is a container that renders divs in a list that
+				can be drag and drop reordered.
+			`,
+			categories: ['controls'],
+		},
+	},
+
+	displayName: 'DraggableList',
+
+	components: {
+		Item: createClass({
+			displayName: 'DraggableList.Item',
+			statics: {
+				peek: {
+					description: `
+						Renders a \`<div>\` that acts as an item in the list
+					`,
+				},
+			},
+			propName: 'Item',
+		}),
+	},
+
+	reducers,
+
+	propTypes: {
+		className: string`
+			Appended to the component-specific class names set on the root element.
+		`,
+		style: object`
+			Passed through to the root element.
+		`,
+		hasDragHandle: bool`
+			Render a drag handle on list items
+		`,
+		dragIndex: number`
+			Index of the item the drag was started on
+		`,
+		dragOverIndex: number`
+			Index of the item the dragged item is hovered over
+		`,
+		onDragStart: func`
+			Called when the user starts to drag an item.
+			Signature: \`(dragIndex, { event, props }) => {}\`
+		`,
+		onDragEnd: func`
+			Called when the user stops to dragging an item.
+			Signature: \`({ event, props }) => {}\`
+		`,
+		onDragOver: func`
+			Called when the user drags an item over another item.
+			Signature: \`(dragOverIndex, { event, props }) => {}\`
+		`,
+		onDrop: func`
+			Called when the user drops an item in the list
+			Signature: \`({oldIndex, newIndex}, { event, props }) => {}\`
+		`,
+	},
+
+	getDefaultProps() {
+		return {
+			hasDragHandle: true,
+			onDragStart: _.noop,
+			onDragEnd: _.noop,
+			onDragOver: _.noop,
+			onDrop: _.noop,
+		};
+	},
+
+	isValidDropIndex(index) {
+		const { dragIndex } = this.props;
+		return index < dragIndex || index > dragIndex + 1;
+	},
+
+	handleDragStart(index) {
+		return event => {
+			const { dataTransfer } = event;
+			dataTransfer.effectAllowed = 'move';
+			dataTransfer.dropEffect = 'move';
+			dataTransfer.setData('drag', 'drag');
+			this.props.onDragStart(index, { event, props: this.props });
+		};
+	},
+
+	handleDragEnd(event) {
+		const { dragIndex, dragOverIndex, onDragEnd, onDrop } = this.props;
+		onDragEnd({ event, props: this.props });
+		if (this.isValidDropIndex(dragOverIndex)) {
+			onDrop(
+				{
+					oldIndex: dragIndex,
+					newIndex:
+						dragOverIndex > dragIndex ? dragOverIndex - 1 : dragOverIndex,
+				},
+				{ event, props: this.props }
+			);
+		}
+	},
+
+	handleDragOver(index) {
+		return event => {
+			event.preventDefault();
+			const { dragOverIndex, onDragOver } = this.props;
+			if (dragOverIndex !== index) {
+				onDragOver(index, { event, props: this.props });
+			}
+		};
+	},
+
+	handleDragLeave(event) {
+		const { dragIndex, onDragOver } = this.props;
+		const childCount = findTypes(this.props, DraggableList.Item).length;
+
+		if (this.lastItemEl) {
+			const { bottom } = this.lastItemEl.getBoundingClientRect();
+			if (_.isNumber(dragIndex) && event.clientY > bottom) {
```

Comment:
if we're ever worried about guarding against NaN in these isNumber checks we could switch to isFinite

Reply:
Sure, I'll change it.